Refactor add to playlist modal

kgarner7 opened the pull request:
The goal of this PR is to refactor add to playlist modal to support the following features:
- Ability to create multiple new playlists
- Checkbox select table to select playlists
- Search table to search playlists.

Inspired heavily from the implementation that Navidrome uses.
This also fixes the problem where the select dropdown opens too early.
 
Resolves #43

kgarner7 committed:
Refactor add to playlist modal

jeffvli commented on src/renderer/features/playlists/components/add-to-playlist-context-modal.tsx:
```diff
@@ -1,6 +1,7 @@
-import { useForm } from '@mantine/form';
+import { Pill } from '@mantine/core';
```
Wrap the component instead of importing from @mantine/core directly

jeffvli commented:
I'm going to hijack this PR and add a couple styling changes to it.

jeffvli committed:
redesign base modal component, add ModalButton component

jeffvli committed:
remove user-select from modal title

jeffvli committed:
update styles for previous modals

kgarner7 committed:
move pill to its own component, fix spelling

jeffvli committed:
set pill default size to md

jeffvli committed:
fix props on Image component

jeffvli committed:
improve visibility of filled button focus

jeffvli committed:
standardize pill styling

jeffvli committed:
optimize add to playlist modal

- add autofocus to the search bar on open
- add keyboard accessibility
- use form to handle submit event
- better distinction between new and existing playlist in pills
- move new playlist to button instead of table row
- update table row styling to include img/description

jeffvli committed:
replace playlist description with other metadata

jeffvli committed:
fix focus visible on filled button
